src/game/creature.c

```diff
@@ -706,3 +711,80 @@ bool Creature_ShootAtLara(
 
     return hit;
 }
+
+bool Creature_TestHybridState(
+    int16_t item_num, int32_t *wh, const HYBRID_INFO *info)
+{
+    // Test the environment for a hybrid creature. Record the water height and
+    // return whether or not a type conversion has taken place.
+    ITEM_INFO *item = &g_Items[item_num];
+    return item->object_number == info->land.id
+        ? Creature_TestLandState(item_num, wh, info)
+        : Creature_TestWaterState(item_num, wh, info);
+}
```

Comment:
The name `Test` suggests that the function doesn't mutate the global state, while it changes up item animations and object numbers. Furthermore `Hybrid` is unclear until the reader already knows the concept behind it. How about `Creature_EnsureHabitat` or similar?

Reply:
Good point, thanks. Now renamed as above, and the statics have been renamed similarly too. I've also refactored slightly to call `Room_GetWaterHeight` in the wrapper function rather than in each state function for a little less repetition.